keps/sig-node/4576-conditions-for-pod-terminated/README.md

```diff
@@ -0,0 +1,906 @@
+`PodTerminated` condition in pod status to indicate a pod has
+reached a state where it's containers have terminated.
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+There are two phases, Succeeded and Failed, which are used to indicate that 
+all containers in the pod have been terminated.  
+When all containers of the pod are terminated, the kubelet will update the phase of 
+the pod to Succeeded or Failed in the SyncPod,  in which case it is not safe to remove the corresponding resource dependencies 
+such as pvc protection, because the phase of the pod may already be in the Succeeded or Failed state 
+but the volumes are not unmounted(the umount process is asynchronous with the pod's phase change).
+
+A new dedicated condition `PodTerminated` indicate that
+all containers in the pod have been terminated and corresponding resource dependencies have been processed
+- volumes are unmounted
+- secrets are unregistered
+- configmaps are unregistered
+- cgroups are removed
+- userns are released
+
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+--> 
+Custom pod controllers/operators can use a dedicated condition `PodTerminated` indicating
+a pod has
+reached a state where it's containers have terminated and corresponding resource dependencies have been processed.
+
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+##### Story 1: Remove pvc protection when pod reaches a completed state
+The PVC can be safely deleted when a job that mounts a PVC reaches a completed state.
+```
+status:
+  conditions:
+  ...
+  - lastProbeTime: null
+    lastTransitionTime: "2024-04-14T15:33:47Z"
+    status: "False"
+    type: PodTerminated
+  - lastProbeTime: null
+    lastTransitionTime: "2024-04-14T15:33:46Z"
+    status: "True"
+    type: PodScheduled
+```
+
```

Comment:
Again, this KEP reads that we are proposing a condition. External controllers could use this condition but do you want to consider using this to solve the user story you had in mind?

Reply:
Updated.